skycatalogs/trilegal_catalog_creator.py

```
@@ -0,0 +1,446 @@
+        hp             int             the healpixel
+        arrow_schema   pyarrow.schema  schema to use
+
+        Returns
+        -------
+        Number of files written (0 or 1)
+
+        '''
+
+        MAX_QUERY_ROWS = 1_000_000
+        #   MAX_QUERY_ROWS = 100_000            # temp for testing
+
+        # Note dl is not part of desc-python or LSST Pipelines; it must be
+        # separately installed
+        from dl import queryClient as qc
+        _NSIDE = 32
+
+        # Form queries and issue
+        nrows = get_trilegal_hp_nrows(hp, nside=_NSIDE)
+        out_nside, out_ring, query_pixels = find_trilegal_subpixels(hp, nrows)
+        n_query = len(query_pixels)
+        to_select = ['ra', 'dec', 'av', 'pmracosd', 'pmdec', 'vrad', 'mu0',
+                     'label as evol_label', 'logte as logT', 'logg',
+                     'logl as logL', 'z as Z',
+                     'umag', 'gmag', 'rmag', 'imag', 'zmag', 'ymag']
+        # all_results = []
+        writer = None
+        rg_written = 0
+        if out_ring:
+            use_column = 'ring256'
+        else:
+            use_column = 'nest4096'
+        # per_query = int(len(query_pixels) / n_query)
+        so_far = 0
+        for iq in range(n_query):
+            in_pixels = ','.join(str(p) for p in query_pixels[iq])
+
+            q = 'select ' + ','.join(to_select)
+
+            q += f' from {self._truth_catalog} where {use_column} in ({in_pixels})'
+            self._logger.debug(f'column {use_column} pixels {in_pixels}')
+            # 600 seconds is max timeout allowed for synchronous query
+            # 300 is generous.  Returning 6 million rows took 80 sec.
+            # Hardly any of these queries return that many rows.
+            try:
+                now = datetime.now().isoformat()[:19]
+                self._logger.debug(f'About to issue query {str(now)}')
+                results = qc.query(adql=q, fmt='pandas', timeout=600)
+            except Exception as e:
```

It looks like you are only interested in exceptions from `qc.query`.  I that case, I'd isolate that line in the `try/except` block:
```
now = datetime.now().isoformat()[:19]
self._logger.debug(f'About to issue query {str(now)}')
try:
    results = qc.query(adql=q, fmt='pandas', timeout=600)
except Exception as e:
    self._logger.debug(f'Exception at time {str(now)}')
    self._logger.debug(str(e))
    raise
```
In general, I'd suggest adding the time stamp to the log messages via the logging config since that's generally useful and to avoid having to compute `now` separately.  Also, if you just let the exception be raised without trying to catch it, the debug log message will tell you when it occurred.

---

Log messages already are configured to include timestamp so, indeed, `now` is not needed.  I'll take it out.